onnxscript/function_libs/torch_aten/param_manipulation.py

```diff
@@ -0,0 +1,171 @@
+"""Function for manipulating input parameters of an Op or a OnnxFunction."""
+from __future__ import annotations
+
+import collections
+import dataclasses
+from typing import Any, List, OrderedDict, Sequence, Tuple
+
+import onnx
+
+from onnxscript import values
+
+# A special value to indicate that the default value is not specified
+_EmptyDefault = object()
+
+
+@dataclasses.dataclass(frozen=True)
+class ParamSchema:
+    """A schema for a parameter of an Op or a OnnxFunction.
+
+    Attributes:
+        name: The name of the parameter.
+        type: The type of the parameter.
+        default: The default value of the parameter.
+        is_input: Whether the parameter is an ONNX input.
+    """
+
+    name: str
+    type: Any = None  # Op input does not have a type, for now
+    default: Any = _EmptyDefault
+    is_input: bool = True
+
+    def __repr__(self) -> str:
```

Comment:
Is the return value supposed to be valid python or just a description?

Reply:
Just wondering whether __str__ is more appropriate or __repr__ ?

Reply:
Good catch, thanks!